config/util/loggers.go

```diff
@@ -0,0 +1,24 @@
+package util
+
+import (
+	"math/rand"
+)
+
+type logMsg func(string, ...interface{})
+
+type randomGenerator func() float32
+
+// LogRandomSample will log a randam sample of the messages it is sent, based on the chance to log
+// chance = 1.0 => always log,
+// chance = 0.0 => never log
+func LogRandomSample(msg string, logger logMsg, chance float32) {
+	logRandomSampleImpl(msg, logger, chance, rand.Float32)
```

Comment:
Locally I added the test that follows to `config/util/loggers_test.go` and it failed with the error message shown below. Do we need to call `rand.New` from inside `LogRandomSample(msg string, logger logMsg, chance float32)` too? Can we add a test to cover this function?
```
func TestLogRandomSampleWrapper(t *testing.T) {
	var buf bytes.Buffer

	mylogger := func(msg string, args ...interface{}) {
		buf.WriteString(fmt.Sprintf(fmt.Sprintln(msg), args...))
	}

	LogRandomSample("This is test line 1", mylogger, 0.5)
	LogRandomSample("This is test line 2", mylogger, 0.5)
	LogRandomSample("This is test line 3", mylogger, 0.5)
	LogRandomSample("This is test line 4", mylogger, 0.5)
	LogRandomSample("This is test line 5", mylogger, 0.5)

	assert.EqualValues(t, expected, buf.String())
}
```
**Result**
```
config/util/loggers_test.go|46| 
|| 	Error Trace:	loggers_test.go:46
|| 	Error:      	Not equal: 
|| 	            	expected: "This is test line 2\nThis is test line 3\n"
|| 	            	actual  : "This is test line 4\nThis is test line 5\n"
|| 	            	
|| 	            	Diff:
|| 	            	--- Expected
|| 	            	+++ Actual
|| 	            	@@ -1,3 +1,3 @@
|| 	            	-This is test line 2
|| 	            	-This is test line 3
|| 	            	+This is test line 4
|| 	            	+This is test line 5
|| 	            	 
|| 	Test:       	TestLogRandomSampleWrapper
```